vpr/src/pack/re_cluster_util.h

```diff
@@ -107,7 +108,8 @@ bool pack_mol_in_existing_cluster(t_pack_molecule* molecule,
                                   std::unordered_set<AtomBlockId>* new_clb_atoms,
                                   bool during_packing,
                                   t_clustering_data& clustering_data,
-                                  t_lb_router_data*& router_data);
+                                  t_lb_router_data*& router_data,
+                                  const int detailed_routing_stage = 1);
```

Comment:
This argument of the function does not match the declaration. The implementation passes the int by reference and this is saying to pass by value. Line 91 above also has this issue. This is causing warnings in the CI and we fail on warning now:
<img width="1057" alt="image" src="https://github.com/verilog-to-routing/vtr-verilog-to-routing/assets/49374526/0c4556ab-bdff-430c-a65f-ced36f1cd4b3">

This *should* actually crash; however, since there is no code currently in VTR using this method, it does not cause an issue. But the warning is saving us here. This should be fixed.

Reply:
Fixed.  Thanks for finding this.